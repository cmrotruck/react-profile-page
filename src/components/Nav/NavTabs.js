import React from "react";
import "bootstrap/dist/css/bootstrap.min.css";

// Here we are using object destructuring assignment to pluck off our variables from the props object
// We assign them to their own variable names
function NavTabs({ currentPage, handlePageChange }) {
  return (
    <ul className="nav justify-content-end">
      <li className="nav-item">
        <a
          href="#aboutMe"
          onClick={() => handlePageChange("aboutMe")}
          // This is a conditional (ternary) operator that checks to see if the current page is "Home"
          // If it is, we set the current page to 'nav-link-active', otherwise we set it to 'nav-link'
          className={
            currentPage === "aboutMe" ? "nav-link active item" : "nav-link item"
          }
        >
          About Me
        </a>
      </li>
      <li className="nav-item">
        <a
          href="#Projects"
          onClick={() => handlePageChange("Projects")}
          // Check to see if the currentPage is `About`, and if so we use the active link class from bootstrap. Otherwise, we set it to a normal nav-link
          className={
            currentPage === "Projects"
              ? "nav-link active item"
              : "nav-link  item"
          }
        >
          Projects
        </a>
      </li>
      <li className="nav-item">
        <a
          href="#Contact"
          onClick={() => handlePageChange("Contact")}
          // Check to see if the currentPage is `Blog`, and if so we use the active link class from bootstrap. Otherwise, we set it to a normal nav-link
          className={
            currentPage === "Contact"
              ? "nav-link active item"
              : "nav-link  item"
          }
        >
          Contact
        </a>
      </li>
      <li className="nav-item">
        <a
          href="#Resume"
          onClick={() => handlePageChange("Resume")}
          // Check to see if the currentPage is `Contact`, and if so we use the active link class from bootstrap. Otherwise, we set it to a normal nav-link
          className={
            currentPage === "Resume" ? "nav-link active item" : "nav-link  item"
          }
        >
          Resume
        </a>
      </li>
    </ul>
  );
}

export default NavTabs;
