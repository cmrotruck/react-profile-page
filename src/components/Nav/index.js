import React from "react";

const Nav = () => {
  return (
    <header>
      <div className="ui vertical masthead center aligned segment">
        <div className="ui container">
          <div className="ui large secondary pointing menu">
            <div className="left item">
              <h1 href="/" className="ui">
                Chad Rotruck
              </h1>
            </div>
            <a href="/" className="item">
              About Me
            </a>
            <a href="/" className="item">
              Portfolio
            </a>
            <a href="/" className="item">
              Resume
            </a>
            <a href="/" className="item">
              Contact Me
            </a>
          </div>
        </div>
      </div>
    </header>
  );
};

export default Nav;
