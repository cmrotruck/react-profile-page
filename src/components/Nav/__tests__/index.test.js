import React from "react";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import Nav from "..";

afterEach(cleanup);

describe("Nav component", () => {
  it("renders correctly", () => {
    render(<Nav />);
  });

  it("matches snapshot", () => {
    const { asFragment } = render(<Nav />);

    expect(asFragment()).toMatchSnapshot();
  });
});
