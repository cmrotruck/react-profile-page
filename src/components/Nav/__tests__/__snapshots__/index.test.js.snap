// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav component matches snapshot 1`] = `
<DocumentFragment>
  <header
    class="flex-row px-1"
  >
    <h1>
      <a
        href="/"
      >
        Chad Rotruck
      </a>
    </h1>
    <nav>
      <ul
        class="flex-row"
      >
        <li
          class="mx-2"
        >
          About Me
        </li>
      </ul>
    </nav>
  </header>
</DocumentFragment>
`;
